lib/host/wasi_nn/wasinnfunc.cpp

```diff
@@ -3,49 +3,478 @@
+      return static_cast<uint32_t>(WASINN::ErrNo::Busy);
+    }
+    Status = ie_blob_size(InputBlob, &BlobSize);
+    spdlog::debug("[WASI-NN] Blob size {}, with Tensor size {}", BlobSize,
+                  TensorDataLen / 4);
+    Status = ie_blob_get_cbuffer(InputBlob, &BlobCBuffer);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error("[WASI-NN] Unable to find input tensor buffer");
+      ie_blob_free(&InputBlob);
+      ie_network_name_free(&InputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::MissingMemory);
+    }
+
+    float *BlobData =
+        const_cast<float *>(static_cast<float const *>(BlobCBuffer.cbuffer));
+    float *CastedTensorDataBuf = reinterpret_cast<float *>(TensorDataBuf);
+    for (size_t I = 0; I < (TensorDataLen / 4); I++) {
+      BlobData[I] = CastedTensorDataBuf[I];
+    }
+    Status = ie_infer_request_set_blob(InferRequest, InputName, InputBlob);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error("[WASI-NN] Unable to set input tensor to model correctly: "
+                    "erro code {}",
+                    Status);
+      ie_blob_free(&InputBlob);
+      ie_network_name_free(&InputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::Busy);
+    }
+    ie_blob_free(&InputBlob);
+    ie_network_name_free(&InputName);
+
+    return static_cast<uint32_t>(WASINN::ErrNo::Success);
+#else
+    spdlog::error("[WASI-NN] OpenVINO backend is not built, use "
+                  "-DWASMEDGE_WASINN_BUILD_OPENVINO=ON"
+                  "to build it.");
+#endif
+  } else {
+    spdlog::error("[WASI-NN] Current backend is not supported.");
+  }
+  return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
 }
 
-Expect<uint32_t> WasiNNGetOuput::body(
-    Runtime::Instance::MemoryInstance *MemInst [[maybe_unused]],
-    uint32_t Context [[maybe_unused]], uint32_t Index [[maybe_unused]],
-    uint32_t OutBuffer [[maybe_unused]],
-    uint32_t OutBufferMaxSize [[maybe_unused]],
-    uint32_t BytesWrittenPtr [[maybe_unused]]) {
-  return 0;
+Expect<uint32_t>
+WasiNNGetOuput::body(Runtime::Instance::MemoryInstance *MemInst,
+                     uint32_t Context, uint32_t Index [[maybe_unused]],
+                     uint32_t OutBuffer [[maybe_unused]],
+                     uint32_t OutBufferMaxSize [[maybe_unused]],
+                     uint32_t BytesWrittenPtr [[maybe_unused]]) {
+  if (MemInst == nullptr) {
+    return Unexpect(ErrCode::ExecutionFailed);
+  }
+
+  if (Ctx.GraphContextBackends.size() <= Context) {
+    spdlog::error("[WASI-NN] get_output: Execution Context does not exist");
+    return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+  }
+
+  if (Ctx.GraphContextBackends[Context] == Ctx.BackendsMapping.at("OpenVINO")) {
+#ifdef WASMEDGE_WASINN_BUILD_OPENVINO
+    uint8_t *OutBufferPtr;
+    uint32_t *BytesWritten;
+    OutBufferPtr = MemInst->getPointer<uint8_t *>(OutBuffer, 1);
+    BytesWritten = MemInst->getPointer<uint32_t *>(BytesWrittenPtr, 1);
```

Comment:
As mentioned:

`uint8_t *OutBufferPtr = ...`
`uint32_t *BytesWritten = ...`

And take care, the element count of `MemInst->getPointer<>` may not be 1.